ipaserver/install/ipa_idrange_fix.py

```diff
@@ -0,0 +1,1073 @@
+                self.under1000, ids_out_of_range = separate_under1000(
+                    ids_out_of_range
+                )
+                if len(self.under1000) > 0:
+                    logger.info(
+                        "Found IDs under 1000, which is not recommeneded \
+(if you definitely need ranges proposed for those, use --allowunder1000):"
+                    )
+                    for identity in self.under1000:
+                        logger.info("%s", identity)
+
+            # Get initial divide of IDs into groups
+            groups = group_identities_by_threshold(
+                ids_out_of_range, self.options.rangegap
+            )
+
+            # Get outliers from too small groups and clean groups for
+            # further processing
+            self.outliers, cleangroups = separate_ranges_and_outliers(
+                groups, self.options.minrange
+            )
+
+            # Print the outliers, they have to be moved manually
+            if len(self.outliers) > 0:
+                print()
+                logger.info(
+                    "Identities that don't fit the criteria to get a new \
+range found! Current attributes:\n\
+Minimal range size: %s\n\
+Maximum gap between IDs: %s\n\
+Try adjusting --minrange, --rangegap or move the following identities into \
+already existing ranges:",
+                    self.options.minrange,
+                    self.options.rangegap
+                )
+                for identity in self.outliers:
+                    logger.info("%s", identity)
+
+            if len(cleangroups) > 0:
+                # Get IDrange name base
+                basename = get_rangename_base(self.id_ranges)
+
+                # Create propositions for new ranges from groups
+                for group in cleangroups:
+                    newrange = propose_range(
+                        group,
+                        self.id_ranges,
+                        self.options.ridoffset,
+                        basename,
+                        self.options.norounding,
+                    )
+                    if newrange is not None:
+                        self.propositions_new.append(newrange)
+                        self.id_ranges.append(newrange)
+                        self.id_ranges.sort(key=lambda x: x.first_id)
+            else:
+                print()
+                logger.info(
+                    "No IDs fit the criteria for a new ID range to propose!"
+                )
+
+    def print_intentions(self) -> None:
+        """Function to print out the summary of the proposed changes"""
+        print()
+        logger.info("Summary:")
+
+        if len(self.outliers) > 0:
+            logger.info("Outlier IDs that are too far away to get a range:")
+            for identity in self.outliers:
+                logger.info("%s", identity)
+
+        if self.options.allowunder1000:
+            logger.info("IDs under 1000 were treated like normal IDs.")
```

Comment:
Should this only be displayed if there were actually IDs under 1000 or is the intention to echo that the option is working?

Reply:
Indeed, reordered if logic, now shows only if some ids under 1000 were found, but we have a flag.